lib/galaxy/web/framework/webapp.py

```diff
@@ -630,16 +630,31 @@ def __update_session_cookie(self, name='galaxysession'):
         self.set_cookie(self.security.encode_guid(self.galaxy_session.session_key),
                         name=name, path=self.app.config.cookie_path)
 
+    def check_user_library_import_dir(self, user):
+        if getattr(self.app.config, "user_library_import_dir_auto_creation", False):
+            # try to create a user library import directory
+            try:
+                self.app.config._ensure_directory(os.path.join(self.app.config.user_library_import_dir, user.email))
```

Comment:
Calling this from outside of the config object seems strange, maybe this should use `galaxy.util.path.safe_makedirs`?